### DISCUSSION TITLE
Component with public property with Collection type

### DISCUSSION THREAD

**Discussion by lukasleitsch** [wire-elements/modal#388] [Q&A] [answered]
A Modal Component with a Collection as public property can not be created and fails with `Method Illuminate\Support\Collection::resolveRouteBinding does not exist.` in https://github.com/wire-elements/modal/blob/0f64fc4c4be49a343a6e025d81629113d5ccdc93/src/Modal.php#L97

```php
class AddNewParticipants extends ModalComponent
{
  public Collection $participants;

  public function mount(array $participants = [])
  {
    $this->participants = User::findMany($participants);
  }
}
```

```html
<button
  wire:click="$dispatch('openModal', { component: 'add-new-participants', arguments: { participants: [2, 4, 5] } })"
>
  @lang('Add')
</x-button>
```

The Modal tries to resolve the parameter `participants` and finds the parameter as public property with the type Collection. It seems that this type is not handled yet.

In this case, I would like to care of resolving the parameters in the mount method and skip the resolving of the Modal Component.

**Comment by PhiloNL** [comment, marked as answer]
I think this is because the argument matches the public property. Try using IDs and computed properties instead, for example:

```php
class AddNewParticipants extends ModalComponent
{
  public $participantsIds;

  #[Computed]
  public function participants()
  {
    return User::findMany($participantsIds);
  }
}
```

```blade
<button
  wire:click="$dispatch('openModal', { component: 'add-new-participants', arguments: { participantsIds: [2, 4, 5] } })"
>
  @lang('Add')
</x-button>
```

**Reply by felixbeer** [reply]
Hey, I ran into the same issue unfortunately.

Wouldn't it make sense to support passing a Collection itself in the first place? 
For me it's the following example:
```php
wire:click="$dispatch('openModal', { component: 'media-manager', arguments: { selectedMedia: {{collect($selectedMedia)}} } })"
```

``selectedMedia`` is a public Collection which contains a [laravel-data](https://github.com/spatie/laravel-data) class in each element. So the ``collect($selectedMedia)``resolves to:
```json
[{"id": 75, "variant": "optimized"}, {"id": 76, "variant": "thumbnail"}]
```

As a result I also get the same error: ``Method Illuminate\Support\Collection::resolveRouteBinding does not exist.``

Maybe the [``resolveParameters()``](https://github.com/wire-elements/modal/blob/d6dac3a6ef6527a734504d8a84de49ec0a72282b/src/Modal.php#L79) method could add an extra case for collections before trying to resolve as a model? 
Would be glad to open a PR for that!
